bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetLiteHandler.java

```diff
@@ -0,0 +1,187 @@
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EchonetLiteHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Barker - Initial contribution
+ */
+@NonNullByDefault
+public class EchonetLiteHandler extends BaseThingHandler implements EchonetDeviceListener {
+    private final Logger logger = LoggerFactory.getLogger(EchonetLiteHandler.class);
+
+    private @Nullable InstanceKey instanceKey;
+    private final Map<String, State> stateByChannelId = new HashMap<>();
+
+    public EchonetLiteHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Nullable
+    private EchonetLiteBridgeHandler bridgeHandler() {
+        @Nullable
+        final Bridge bridge = getBridge();
+        if (null == bridge) {
+            logger.warn("Unable to handle commands if bridge is null");
+            return null;
+        }
+
+        @Nullable
+        final EchonetLiteBridgeHandler handler = (EchonetLiteBridgeHandler) bridge.getHandler();
+        if (null == handler) {
+            logger.warn("Unable to handle commands if bridge handler is null");
+            return null;
+        }
+
+        return handler;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        @Nullable
+        final EchonetLiteBridgeHandler handler = bridgeHandler();
+        if (null == handler) {
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            logger.debug("Refreshing: {}", channelUID);
+
+            final State currentState = stateByChannelId.get(channelUID.getId());
+            if (null == currentState) {
+                handler.refreshDevice(requireNonNull(instanceKey), channelUID.getId());
+            } else {
+                updateState(channelUID, currentState);
+            }
+        } else if (command instanceof State) {
+            logger.debug("Updating: {} to {}", channelUID, command);
+
+            handler.updateDevice(requireNonNull(instanceKey), channelUID.getId(), (State) command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final EchonetDeviceConfig config = getConfigAs(EchonetDeviceConfig.class);
+
+        logger.info("Initialising: {}", config);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        @Nullable
+        final EchonetLiteBridgeHandler bridgeHandler = bridgeHandler();
+        if (null == bridgeHandler) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Bridge is null");
```

Comment:
Please support localization by adding an i18n string here. See for example:
https://github.com/openhab/openhab-addons/blob/e4917c3b4a41a178c5176b75d786dfd383c807ee/bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java#L109-L110

and:
https://github.com/openhab/openhab-addons/blob/e4917c3b4a41a178c5176b75d786dfd383c807ee/bundles/org.openhab.binding.hdpowerview/src/main/resources/OH-INF/i18n/hdpowerview.properties#L70